File: src/main/org/nlogo/api/FileIO.java
package org.nlogo.api;

public final strictfp class FileIO {

  // this class is not instantiable
  private FileIO() {
    throw new IllegalStateException();
  }

  public static String file2String(String path)
      throws java.io.IOException {
    java.io.FileReader fr = new java.io.FileReader(path);
    return org.nlogo.util.Utils.reader2String(fr);
  }

  public static void writeFile(String path, String text)
      throws java.io.IOException {
    writeFile(path, text, false);
  }

  public static void writeFile
      (String path, String text, boolean convertToPlatformLineBreaks)
      throws java.io.IOException {
    org.nlogo.api.File file = null;
    try {
      file = new org.nlogo.api.LocalFile(path);
      file.open(org.nlogo.api.File.Mode.WRITE);
      if (!convertToPlatformLineBreaks) {
        file.print(text);
      } else {
        java.io.BufferedReader lineReader = new java.io.BufferedReader(
            new java.io.StringReader(text));
        String line = lineReader.readLine();
        while (line != null) {
          file.println(line);
          line = lineReader.readLine();
        }
      }
      file.close(true);
    } finally {
      if (file != null) {
        file.close(false);
      }
    }
  }
}

File: src/main/org/nlogo/api/FileIO.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.api

object FileIO {

  @throws(classOf[java.io.IOException])
  def file2String(path: String) =
    org.nlogo.util.Utils.reader2String(
      new java.io.FileReader(path))

  @throws(classOf[java.io.IOException])
  def writeFile(path: String, text: String) {
    writeFile(path, text, false)
  }

  @throws(classOf[java.io.IOException])
  def writeFile(path: String, text: String, convertToPlatformLineBreaks: Boolean) {
    val file = new LocalFile(path)
    try {
      file.open(FileMode.Write)
      if (!convertToPlatformLineBreaks)
        file.print(text)
      else {
        val lineReader = new java.io.BufferedReader(
          new java.io.StringReader(text))
        val lines =
          Iterator.continually(lineReader.readLine()).takeWhile(_ != null)
        for(line <- lines)
          file.println(line)
      }
      file.close(true)
    }
    finally file.close(false)
  }

}
